src/main/java/com/linkedin/xinfra/monitor/services/MultiClusterTopicManagementService.java

```diff
@@ -416,90 +415,91 @@ int numPartitions() throws InterruptedException, ExecutionException {
+        LOGGER.debug(
+            "Configured replication factor {} is smaller than the current replication factor {} of the topic {} in cluster.",
+            _replicationFactor, currentReplicationFactor, _topic);
+      }
 
-        if (expectedReplicationFactor > currentReplicationFactor && Utils.ongoingPartitionReassignments(_adminClient)
-            .isEmpty()) {
-          LOGGER.info(
-              "MultiClusterTopicManagementService will increase the replication factor of the topic {} in cluster"
-                  + "from {} to {}", _topic, currentReplicationFactor, expectedReplicationFactor);
-          reassignPartitions(_adminClient, brokers, _topic, partitionInfoList.size(), expectedReplicationFactor);
+      if (expectedReplicationFactor > currentReplicationFactor && Utils.ongoingPartitionReassignments(_adminClient)
+          .isEmpty()) {
+        LOGGER.info("MultiClusterTopicManagementService will increase the replication factor of the topic {} in cluster"
+            + "from {} to {}", _topic, currentReplicationFactor, expectedReplicationFactor);
+        reassignPartitions(_adminClient, brokers, _topic, partitionInfoList.size(), expectedReplicationFactor);
+        partitionReassigned = true;
+      }
 
-          partitionReassigned = true;
-        }
+      // Update the properties of the monitor topic if any config is different from the user-specified config
+      Properties currentProperties = new Properties(); // ConfigResource -> Collection<AlterConfigOp>
+      Properties expectedProperties = new Properties(); // ConfigResource -> Collection<AlterConfigOp>
+      ConfigResource configResource = new ConfigResource(ConfigResource.Type.TOPIC, _topic);
+      DescribeConfigsResult describeConfigsResult =
+          _adminClient.describeConfigs(Collections.singleton(configResource));
 
-        // Update the properties of the monitor topic if any config is different from the user-specified config
-        Properties currentProperties = zkClient.getEntityConfigs(ConfigType.Topic(), _topic);
-        Properties expectedProperties = new Properties();
-        for (Object key : currentProperties.keySet()) {
-          expectedProperties.put(key, currentProperties.get(key));
-        }
-        for (Object key : _topicProperties.keySet()) {
-          expectedProperties.put(key, _topicProperties.get(key));
-        }
-
-        if (!currentProperties.equals(expectedProperties)) {
-          LOGGER.info("MultiClusterTopicManagementService will overwrite properties of the topic {} "
-              + "in cluster from {} to {}.", _topic, currentProperties, expectedProperties);
-          zkClient.setOrCreateEntityConfigs(ConfigType.Topic(), _topic, expectedProperties);
-        }
+      for (ConfigEntry configEntry : describeConfigsResult.values().get(configResource).get().entries()) {
+        currentProperties.put(configResource, new AlterConfigOp(configEntry, AlterConfigOp.OpType.SET));
```

Comment:
`currentProperties` will only contains 1 entry with the last `ConfigEntry` as it's putting always with the same unique key `configResource`.

Take a look at PR #319 that does the job

Reply:
Approved and merged your PR including these changes.
Seeking your +1 on this PR for minor modifications